docs/implementation-guide.md

```diff
@@ -0,0 +1,122 @@
+
+If you do not have meet either of these criteria contact the owner of your organization's GCP environment.
+
+### Step 2: Install the Gcloud CLI.
+The gcloud CLI is a command line tool provided by Google for working with GCP. We will use it to authenticate your local machine with your organization's GCP environment. Follow [this guide](https://cloud.google.com/sdk/docs/install) to install gcloud. After installation run `gcloud auth application-default login` and follow the prompts in your browser to login to GCP.
+
+### Step 3: Fork the phdi-google-cloud Respository
+Fork the phdi-google-cloud respository into your organization's, or your personal, GitHub account.
+1. Navigate to [https://github.com/CDCgov/phdi-google-cloud](https://github.com/CDCgov/phdi-google-cloud).
+2. Click on the `Fork` button in the top right.
+![fork-repo-1](./fork-repo-1.png)
+3. Select the owner of the forked repository. We recommend that this is an organization and not an individual.
+4. Click `Create fork` at the bottom of the page.
+![fork-repo-2](./fork-repo-2.png)
+
+### Step 4: Clone the Forked Repository
+Clone the forked version of the phdi-google-cloud repository by running `git clone https://github.com/<MY-GITHUB-ORGANIZATION>/phdi-google-cloud.git`. If you do not have `git` installed please follow [this guide](https://github.com/git-guides/install-git) to install it.
+
+### Step 5: Run the Quick Start Script
+In this step we will work through GCP's [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) to granted your phdi-google-cloud repo access to deploy the pipelines to your organization's GCP environment. We have provided a script to automate most of this process that we recommend you use. However if you prefer to work through it manually you may follow [this guide](https://github.com/google-github-actions/auth#setup).
+
+From your machine's command line:
+1. Navigate to the root directory of the repository you cloned in step 4.
+2. Login into GCP by running `gcloud auth application-default login`.
+3. Run the quick start script and follow the prompts.
+    - [quick-start.sh](../quick-start.sh) for Mac an Linux
+    - TODO: WRITE WINDOWS BATCH QUICK START SCRIPT
+
+If you plan to deploy to an existing project in your GCP environment, have the project ID ready and provided it to the quick start script when prompted.
+
+### Step 6: Set Repository Secrets
+Set the following secret values in your forked phdi-google-cloud repository:
+- `PROJECT_ID` - Specified by the quick start script.
+- `SERVICE_ACCOUNT_ID` - Specified by the quick start script.
+- `WORKLOAD_IDENTITY_PROVIDER` - Specified by the quick start script.
+- `REGION` - Your choice of GCP region.
+- `ZONE`- Your choice of GCP zone.
+
+Information about GCP regions and zones is available [here](https://cloud.google.com/compute/docs/regions-zones).
+
+To create a repository secret follow this steps.
```

Comment:
I'm a little confused - I thought the quick start script set these environment variables in Github. Is this only if you want to manually set them instead of using the quickstart script? Or does that startup script only set the key and you need to specify the value in the UI?

Reply:
The quick start script automates the [Workload Identity Federation process](https://github.com/google-github-actions/auth#setup) to authenticate GitHub with GCP. Importantly, it produces values for `SERVICE_ACCOUNT_ID`, `WORKLOAD_IDENTITY_PROVIDER`, as well as `PROJECT_ID` if the user choses to create the GCP project via the script. Currently, all of these values still need to be set in the GitHub repo manually. However, it looks like the GitHub CLI tool offers functionality to [set repository secrets](https://cli.github.com/manual/gh_secret_set) with `gh secret set <secret-name> `. This would allow us to include setting the repo secrets in the quick start script, but it would add an additional dependency for running the script, or at least this portion of it. @nickclyde what are your thoughts on this? 

Here is one possible solution:

1. We encourage installation of the GitHub CLI and include it as a step of the implementation guide.
2. After completing the authentication process the script checks for the GitHub CLI.
3. - If the the script is able to `gh` it will set repository secrets for the user and prompt them for the values they must choose
    - If the script can't `gh` it finishes as it is currently written and prompts the user to set the secrets manually.

Reply:
I am in favor of using the GitHub CLI to set secrets! Let's add it as a separate PR though.